src/compiler/crystal/codegen/debug.cr

```diff
@@ -280,9 +471,25 @@ module Crystal
       file, dir = file_and_dir(location.filename)
       scope = di_builder.create_file(file, dir)
       fn_metadata = di_builder.create_function(scope, target_def.name, target_def.name, scope,
-        location.line_number, fun_metadata_type, true, true,
-        location.line_number, LLVM::DIFlags::Zero, false, context.fun)
-      fun_metadatas[context.fun] = fn_metadata
+        location.line_number, fun_metadata_type(context.fun.debug_params), true, true,
+        location.line_number, LLVM::DIFlags::Zero, !@debug.variables?, context.fun)
+      fun_metadatas[context.fun] = [Tuple.new(location.original_filename || "??", fn_metadata)]
+    end
+
+    def create_pointer(debug_type : LibLLVMExt::Metadata, type_name : String)
+      di_builder.create_pointer_type(debug_type, 8u64 * llvm_typer.pointer_size, 8u64 * llvm_typer.pointer_size, type_name)
+    end
+
+    def declare_debug_for_funciton_argument(arg_name, arg_type, arg_no, alloca, location)
```

Comment:
```suggestion
    def declare_debug_for_function_argument(arg_name, arg_type, arg_no, alloca, location)
```

Reply:
Good eye! I didn't even realized at first glance what was wrong ;)

Reply:
Same here! It took me many glances to notice that ;)